File: apps/home-service-platform-node-js-admin/src/api/serviceProvider/ServiceProviderWhereInput.ts
import { StringNullableFilter } from "../../util/StringNullableFilter";
import { BookingWhereUniqueInput } from "../booking/BookingWhereUniqueInput";
import { BookingListRelationFilter } from "../booking/BookingListRelationFilter";
import { FloatNullableFilter } from "../../util/FloatNullableFilter";
import { StringFilter } from "../../util/StringFilter";
import { ReviewWhereUniqueInput } from "../review/ReviewWhereUniqueInput";
import { ReviewListRelationFilter } from "../review/ReviewListRelationFilter";
import { ServiceCategoryListRelationFilter } from "../serviceCategory/ServiceCategoryListRelationFilter";

export type ServiceProviderWhereInput = {
  availability?: StringNullableFilter;
  bio?: StringNullableFilter;
  booking?: BookingWhereUniqueInput;
  bookings?: BookingListRelationFilter;
  email?: StringNullableFilter;
  hourlyRate?: FloatNullableFilter;
  id?: StringFilter;
  name?: StringNullableFilter;
  phone?: StringNullableFilter;
  ratings?: StringNullableFilter;
  review?: ReviewWhereUniqueInput;
  reviews?: ReviewListRelationFilter;
  serviceCategories?: ServiceCategoryListRelationFilter;
  serviceType?: StringNullableFilter;
};

File: apps/home-service-platform-server/src/APIs/ServiceProvider/Dtos/ServiceProviderWhereInput.cs
namespace HomeServicePlatform.APIs.Dtos;

public class ServiceProviderWhereInput
{
    public string? Availability { get; set; }

    public string? Bio { get; set; }

    public string? Booking { get; set; }

    public List<string>? Bookings { get; set; }

    public DateTime? CreatedAt { get; set; }

    public string? Email { get; set; }

    public double? HourlyRate { get; set; }

    public string? Id { get; set; }

    public string? Name { get; set; }

    public string? Phone { get; set; }

    public string? Ratings { get; set; }

    public string? Review { get; set; }

    public List<string>? Reviews { get; set; }

    public List<string>? ServiceCategories { get; set; }

    public string? ServiceType { get; set; }

    public DateTime? UpdatedAt { get; set; }
}
